lib/demos/reference/motion_demo_fade_through_transition.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:animations/animations.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN fadeThroughTransitionDemo
+
+class FadeThroughTransitionDemo extends StatefulWidget {
+  const FadeThroughTransitionDemo();
+
+  @override
+  _FadeThroughTransitionDemoState createState() =>
+      _FadeThroughTransitionDemoState();
+}
+
+class _FadeThroughTransitionDemoState extends State<FadeThroughTransitionDemo> {
+  int pageIndex = 0;
+
+  List<Widget> pageList = [
+    _AlbumsPage(),
+    _PhotosPage(),
+    _SearchPage(),
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(
+          localizations.demoFadeThroughTitle,
+        ),
+      ),
+      body: PageTransitionSwitcher(
+        transitionBuilder: (
+          child,
+          animation,
+          secondaryAnimation,
+        ) {
+          return FadeThroughTransition(
+            animation: animation,
+            secondaryAnimation: secondaryAnimation,
+            child: child,
+          );
+        },
+        child: pageList[pageIndex],
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        currentIndex: pageIndex,
+        onTap: (selectedIndex) {
+          setState(() {
+            pageIndex = selectedIndex;
+          });
+        },
+        items: [
+          BottomNavigationBarItem(
+            icon: const Icon(
+              Icons.photo_library,
```

Comment:
Nit: Usually if it will fit on one line, we remove the trailing comma. Looks like you can do that for the `Icon` and `Text` widgets here and below